src/button-with-icon/types.ts

```diff
@@ -0,0 +1,13 @@
+import React from "react";
+import { ButtonProps } from "../button/types";
+
+export interface ButtonWithIconProps extends ButtonProps {
```

Comment:
If `loading` is not meant to be introduced, then you can't just import from `ButtonProps`